File: leetcode/1029_two_city_scheduling.py
class Solution:
    def twoCitySchedCost(self, costs):
        result = 0
        mid = len(costs) // 2
        costs.sort(key=lambda x: x[0] - x[1])
        for i, (first, second) in enumerate(costs):
            if i < mid:
                result += first
            else:
                result += second
        return result

File: leetcode/java/1029TwoCityScheduling.java
import java.util.Arrays;

class Solution {
  public int twoCitySchedCost(int[][] costs) {
    Arrays.sort(
        costs, (int[] costs1, int[] costs2) -> costs1[0] - costs1[1] - (costs2[0] - costs2[1]));
    int N = costs.length / 2;
    int totalCost = 0;
    for (int i = 0; i < N; i++) {
      totalCost += costs[i][0] + costs[i + N][1];
    }
    return totalCost;
  }
}
